### DISCUSSION TITLE
4:00:26 the part where patrick does the .env and .gitignore using mac/linux.

### DISCUSSION THREAD

**Discussion by slime956** [smartcontractkit/full-blockchain-solidity-course-py#734] [Q&A] [answered]
as you can tell by the title, i have a question. so recently i was taking courses for python and realized everything has to be stored ina folder. ok i got that out of the way and now, i have the ganache-cli, so know im at the step (4:00:26 and after) where Patrick explains the process using mac and linux, he sort of misguided me a little to the point where im all over brave trying to find the solution using windows. 

for example, i cant put ......

# For connecting to ganache
w3 = Web3(Web3.HTTPProvider("http://0.0.0.0:8545"))
chaind_id = 1337
my_address = "0x94B806BB0e455576ea46193D9DBbB08d1cc57Da9"
private_key = os.getenv("PRIVATE_KEY")

......because i dont have os

its the little things , i wish i had a mac in order to proceed with the 16 hour video but i dont and well it all takes time, really wished you can help by showing me how to do it using windows, basically. Im at 4:00:26, barely gonna do what Patrick is doing in mac/linux , but for windows. 

ima go to sleep thanks and take care stay safe.

**Comment by slime956** [comment]
ps i copied and pasted from Pattrick , this is how I have it in MY vs code

# for connecting to ganache
w3 = Web3(Web3.HTTPProvider("http://0.0.0.0:8545"))
chain_id = 1337
my_address = "0x6B25dF9789756e9D38e53C00764D06852874EF6B"
private_key = ("PRIVATE_KEY")
print(private_key)

**Comment by slime956** [comment]
and this is also one thing that bugs me, i put the rpc server theWAY it IS in GANACHE , with the network ID as 5777, but it gave me errors so i just did it that way 

my rpc server is actually HTTP://127.0.0.1:7545
network ID 5777 like i mentioned, i remember i had trouble surpassing this step and when into discussioons for help and found out that it is better to leave it this way. i maybe wrong but honestly ive been trying to get this project done and im only learning, i want to go to a university and study about all this , time is patience.

**Reply by cromewar** [reply]
This is actually because 5777 is not correct (kind of) just change it to 1377.

**Comment by cromewar** [comment, marked as answer]
Hello @slime956, well the process should be the same on any OS you are using, you just need to select the host name ganache gives you, anyways I have some tips for you:

1. Don't use windows.... just a joke, but for programming stuff (specially web3) i Highly recommend you to try WSL (windows subsystem for linux) on the course index "windows support" apart you will have a guide about how to setup it for windows.
2. Don't worry that much, all those connections your are making in web3 lesson are going to be replaced with Brownie network, which manages the connections better.

**Reply by slime956** [reply]
lmfaooo , before i went to bed, i was saying to myself ima just have to learn linux, i got the rufus installed but apparently i need a usb flash drive , im watching this video [[[https://www.youtube.com/watch?v=n_2jPbQornY&t=711s]]] minute 11:49, ima have to go to walmart and buy one, lol. anyways, thank you so much for your advice brother, i really appreciate what you do for the community and all, just wanted to say thanks, stay safe and God bless. ////// if you have any videos or websites related to the link, let me know i would love to learn more.

**Reply by cromewar** [reply]
My pleasure, I develop almost exclusively on windows using WSL, is easier than format your computer and install linux from the bottom.
